tests/integration/Lib/IntegrationTestPagingMultipleBase.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * @author Arthur Schiwon <blizzz@arthur-schiwon.de>
+ * @author Joas Schilling <coding@schilljs.com>
+ *
+ * @copyright Copyright (c) 2016, ownCloud GmbH.
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\User_LDAP\Tests\Integration\Lib;
+
+use OCA\User_LDAP\Tests\Integration\AbstractIntegrationTest;
+use OCA\User_LDAP\Mapping\UserMapping;
+use OCA\User_LDAP\User_LDAP;
+
+require_once __DIR__ . '/../AbstractIntegrationTest.php';
+
+class IntegrationTestPagingMultipleBase extends AbstractIntegrationTest {
+	/** @var  UserMapping */
+	protected $mapping;
+
+	/** @var User_LDAP */
+	protected $backend;
+
+	/**
+	 * prepares the LDAP environment and sets up a test configuration for
+	 * the LDAP backend.
+	 */
+	public function init() {
+		require(__DIR__ . '/../setup-scripts/createExplicitUsers.php');
+		parent::init();
+
+		$this->backend = new \OCA\User_LDAP\User_LDAP(\OC::$server->getConfig(), $this->userManager);
+	}
+
+	/**
+	 * tests that paging having multiple bases returns limit at offset
+	 * for all bases in parallel
+	 *
+	 * @return bool
+	 */
+	protected function case1() {
+		$filter = 'objectclass=inetorgperson';
+		$attributes = ['cn', 'dn'];
+
+		// having multiple bases there is currently an issue that:
+		// - having searchUsers with limit 2 and offset 2, expected resultset is 2
+		// - while having multiple base-dns specified, LDAP currently will return 4 results (while expected is 2 from range query)
+		$result = $this->access->searchUsers($filter, $attributes, 2, 0);
```

Comment:
@jvillafanez @VicDeo @cdamken I might need help here. When having multiple base DNs e.g. `$bdnUsers = 'ou=users,' . $bdn . ';' .'ou=admins,' . $bdn;` , we now return N times results then expected by limit/offset. Is this expected behaviour? Or this means we do not (ownCloud) support specifying multiple base dns per single LDAP server configuration  ?
<img width="964" alt="95899960-94a21000-0d91-11eb-9cd0-4b92ced79aca" src="https://user-images.githubusercontent.com/13368647/96187439-4a5d9200-0f3d-11eb-9903-83bf8f485b8a.png">

Reply:
Either known bug or not supported. Anyway, fixing it would need to rewrite a good amount of LDAP code, and I'm not sure how many people would require to use multiple base dn.
I'd vote for not supported at this point.

Reply:
@micbar option to consider -> add new commit that displays error in UI informing that multi-base DN per single server is currently not supported.